plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -912,6 +914,11 @@ private ConnectorTableMetadata getTableMetadata(ConnectorSession session, Schema
             properties.put(LOCATION_PROPERTY, icebergTable.location());
         }
 
+        if (icebergTable instanceof BaseTable) {
+            int formatVersion = ((BaseTable) icebergTable).operations().current().formatVersion();
+            properties.put(FORMAT_VERSION_PROPERTY, formatVersion);
+        }
```

Comment:
SHOW CREATE TABLE should show this information always, it should always be available.
Instead of `instanceof`, please do an unconditional cast here.


```suggestion
        int formatVersion = ((BaseTable) icebergTable).operations().current().formatVersion();
        properties.put(FORMAT_VERSION_PROPERTY, formatVersion);
```